worker/src/worker.py

```diff
@@ -0,0 +1,154 @@
+
+
+def case_error(case, error):
+    """
+    Set error fields for the given case, and persist it to the database.
+    :param case: The case that has given an error.
+    :param error: The error message.
+    """
+    case['status'] = CaseStatus.ERROR
+    case['error'] = error
+    case_collection.update_case(case['_id'], case)
+
+
+def execute_block(case, block, step):
+    """
+    Executes a block by compiling parameters and calling the correct endpoint on WorkflowBlockService.
+    :param case: The case that is being executed.
+    :param block: The block that should be executed.
+    :param step: The step we are currently on in the case.
+    :return: Results from the block.
+    """
+    block_info = requests.get(block_url + block['name'] + '/info').json()
+
+    params = block['params']
+    insert_params = DotMap({'store': case['store'],
+                            'outputs': case['previous_outputs']})
+    for p_name, p_value in params.items():
+        if type(p_value) == str:
+            # Insert parameters from store and previous outputs.
+            params[p_name] = p_value.format_map(insert_params)
+
+    cleaned_params = {}
+    for p in block_info['params']:
+        try:
+            # Try to cast parameter to correct type.
+            # Required since all data from store or previous step are inserted as strings.
+            typ = type_map[block_info['params'][p]['type']]
+            cleaned_params[p] = typ(params[p])
+        except KeyError:
+            case_error(case, 'Case is missing parameter ' + p + ' required by the block ' +
+                       block['name'] + 'in step ' + step)
+            return None
+        except ValueError:
+            case_error(case, 'Failed to cast parameter value for parameter ' + p + ' in step ' + step)
+            return None
+
+    result = post_json(block['name'], {'params': cleaned_params})
+    return result
+
+
+def save_result(case, result, step_item):
+    case['previous_outputs'] = result['data']
+
+    store = DotMap(case['store'])
+    for saved_output, save_to in step_item['save_outputs'].items():
+        if saved_output in result['data']:
+            path = save_to.split('.')
+            item = store
+            for i in range(len(path) - 1):
+                item = item[path[i]]
+            item[path[-1]] = result['data'][saved_output]
+    case['store'] = store.toDict()
+
+
+def execute_case(case):
+    """
+    Handles the execution of a case, including the execution of single blocks and branching.
+    :param cid: The case ID.
+    """
+    try:
+        workflow = case['workflow']
+        step = case['step']
+
+        while True:
+            step_item = workflow['blocks'][step]
+
+            if step_item['type'] == 'action':
+                print(step)
+                result = execute_block(case, step_item, step)
+
+                if not result:
+                    return
```

Comment:
Could a faulty block interrupt the entire workflow?

Reply:
Yes, and the way I see it, this is also the correct behaviour. If a single block misbehaves, the following blocks will not have access to the expected data.